src/librbd/mirror/snapshot/CreatePrimaryRequest.cc

```diff
@@ -208,11 +241,6 @@ void CreatePrimaryRequest<I>::unlink_peer() {
           info->mirror_peer_uuids.count(peer) == 0) {
```

Comment:
Now that snapshots not linked with any peer are handled separately, I think this condition can be simplified to check just whether the snapshot is linked with this peer:
```
      if (info->mirror_peer_uuids.count(peer) == 0) {
        // snapshot is not linked with this peer
        continue;
      }
```